selfdrive/camerad/cameras/camera_webcam.cc

```diff
@@ -0,0 +1,269 @@
+#include "camera_webcam.h"
+
+#include <unistd.h>
+#include <string.h>
+#include <signal.h>
+#include <pthread.h>
+
+#include "common/util.h"
+#include "common/timing.h"
+#include "common/swaglog.h"
+#include "buffering.h"
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/core.hpp>
+#include <opencv2/videoio.hpp>
+
+extern volatile sig_atomic_t do_exit;
+
+#define FRAME_WIDTH  1164
+#define FRAME_HEIGHT 874
+#define FRAME_WIDTH_FRONT  1152
+#define FRAME_HEIGHT_FRONT 864
+
+namespace {
+void camera_open(CameraState *s, VisionBuf *camera_bufs, bool rear) {
+  assert(camera_bufs);
+  s->camera_bufs = camera_bufs;
+}
+
+void camera_close(CameraState *s) {
+  tbuffer_stop(&s->camera_tb);
+}
+
+void camera_release_buffer(void *cookie, int buf_idx) {
+  CameraState *s = static_cast<CameraState *>(cookie);
+}
+
+void camera_init(CameraState *s, int camera_id, unsigned int fps) {
+  assert(camera_id < ARRAYSIZE(cameras_supported));
+  s->ci = cameras_supported[camera_id];
+  assert(s->ci.frame_width != 0);
+
+  s->frame_size = s->ci.frame_height * s->ci.frame_stride;
+  s->fps = fps;
+
+  tbuffer_init2(&s->camera_tb, FRAME_BUF_COUNT, "frame", camera_release_buffer, s);
+}
+
+static void* rear_thread(void *arg) {
+  int err;
+
+  set_thread_name("webcam_rear_thread");
+  CameraState* s = (CameraState*)arg;
+
+  cv::VideoCapture cap_rear(1); // road
+  cap_rear.set(cv::CAP_PROP_FRAME_WIDTH, 1280);
+  cap_rear.set(cv::CAP_PROP_FRAME_HEIGHT, 720);
+  cap_rear.set(cv::CAP_PROP_FPS, s->fps);
```

Comment:
Does this set the framerate of the camera?

Reply:
Yes. Found out this can actually limit the model fps